app/javascript/dashboard/store/modules/conversations/index.js

```diff
@@ -85,17 +85,19 @@ export const mutations = {
       selectedChatId: conversationId,
     });
     if (!chat) return;
-    const previousMessageIndex = chat.messages.findIndex(
-      m => m.id === message.id
+    const { echo_id: tempMessageId } = message;
+    const pendingMessageIndex = chat.messages.findIndex(
+      m => m.id === message.id || m.id === tempMessageId
```

Comment:
`m.echo_id === tempMessageId`?

Reply:
`echo_id` is just o get the echo, we are keeping the temp value in `id` itself. Else we will have clear echo_id and check each time as there is response from both `action-cable` and `api`